src/main/java/net/openhft/chronicle/queue/impl/single/StoreTailer.java

```diff
@@ -195,6 +201,25 @@ public DocumentContext readingDocument(final boolean includeMetaData) {
         return documentContext;
     }
 
+    DocumentContext readingDocumentNamed(final boolean includeMetaData) {
+        for (int i = 0; i < 100; i++) {
+            this.indexChecker = indexValue.getVolatileValue();
+            if (this.index != indexChecker)
+                moveToIndex(this.indexChecker);
+
+            DocumentContext documentContext = readingDocument0(includeMetaData);
+
+            if (indexChecker != Long.MIN_VALUE) {
+                this.index = indexChecker;
+                if (context.isPresent() && !context.isMetaData())
+                    incrementIndex();
+                return documentContext;
+            }
+            documentContext.close();
+        }
+        throw new AssertionError();
```

Comment:
Is it possible in cases of high contention that you'd get to the end of the 100 iterations? Probably very unlikely but even if it's a remote possibility we should probably put a meaningful exception/message in here.